### DISCUSSION TITLE
将当前阅读模式转换为 Telegraph Page 并发送到你的 Telegram Channel

### DISCUSSION THREAD

**Discussion by Kenshin** [Kenshin/simpread#3337] [服务]
> 此功能来自 [简悦社区](https://t.me/simpread) 用户 @OverflowCat

> 如果你在使用 Telegram Channel 作为获取信息的渠道，可以试试此方式。

https://user-images.githubusercontent.com/81074/150478215-575d9928-1a93-48f3-b320-8ec33b7492aa.mp4

# 特点

1. Telegraph Page 在 Telegram Mobile 端可以直接使用 Instant view 浏览方式
2. 导入 Telegraph Page 的图片使用了 `telegra.ph` 图床
3. 这套方式可以融入到自动化，即：加入稍后读后自动导出到你的 Telegram Channel
4. 接入 Telegram Channel 后，可以将其转换为 RSS，用以弥补简悦 → 稍后读没有 RSS 的「问题」

# 配置难度

简单，无需使用同步助手，且在 [Kiwi Browser](https://zhuanlan.zhihu.com/p/415622477) 上也可以使用。

# 如何配置

1. 添加 简悦 · 导出助手 小二 [@simpread_send_bot](https://t.me/simpread_send_bot) 到你的 Telegram Channel
2. 赋予 简悦 · 导出助手 管理员权限

3. 将下面的代码加入到 Webhook（选项页 → 服务 → Webhook）

```
{
    "name": "Webhook2Telegraph",
    "url": "https://api-wrap-2.simpread.pro/api/service/telegraph",
    "type": "POST",
    "headers": {
        "Content-Type": "application/json"
    },
    "body": {
        "article": {
            "url": "{{url}}",
            "title": "{{title}}",
            "desc": "{{desc}}",
            "tags": "{{tags}}",
            "content": "{{content}}"
        },
        "telegram": {
            "bind": true,
            "link": "你的 Telegram Channel",
            "name": "你的名字"
        }
    }
}
```

- `你的 Telegram Channel` → `https://t.me/simpread_rss` （如果是 Private Channel 请往下看）
- `你的名字` → `Kenshin`

4. 将上面的代码使用 https://tool.lu/json/ 提供的压缩方式压缩一行，粘贴到 **选项页 → 服务 → Webhook**

![image](https://user-images.githubusercontent.com/81074/150479714-82bca6e1-d68f-48c5-890f-048e0846a9ae.png)

✅ Done

# 如何使用

## 自动触发（加入稍后读后）

将 `Webhook2Telegraph` 加入到 **选项页 → 服务 → Webhook**

![image](https://user-images.githubusercontent.com/81074/150479867-c9784faf-e73a-4c75-8694-ae0352d45461.png)

## 手动触发

进入阅读模式后，**右下角 → 动作 → 保存 → Webhook2Telegraph**

![image](https://user-images.githubusercontent.com/81074/150480366-030c95d1-6810-480f-a5e7-be3e7a968d2e.png)

# 高级设置

可以使用自己的 Telegram bot 服务替代  [简悦 · 导出助手 小二](https://t.me/simpread_send_bot) 配置如下

```
"telegram": {
    ... ...
    "bot": "你自己的 Telegram bot"
}
```

可以使用自己的 Telegraph Page Token 配置如下

```
"telegraph": {
    "token": "你自己的 Telegraph token"
}
```

**注意：**使用自己的 Telegraph Page Token 可管理全部通过此方式生成的 Telegraph 页面，否则将会使用内置的 Token

以下是全部配置

![image](https://user-images.githubusercontent.com/81074/150481239-87abe1aa-7c1e-4930-b2d8-5933d05589ef.png)

# 如何获取 Channel

如果是 Public Channel 的话，请使用上面提到的 `link` 字段，如果是 Prviate Channle 的话，请使用下面的方式

1. 添加 [@username_to_id_bot](http://t.me/username_to_id_bot)

2. 将 Private Channel 链接发送给它

![image](https://user-images.githubusercontent.com/81074/150497120-7c3c8b01-281a-4ec0-9cd6-3308bc3a7103.png)

3. 在 `telegram` 增加 `chat` 字段，并按下图所示设置

![image](https://user-images.githubusercontent.com/81074/150505066-3edaf4d2-5814-48f3-852d-21839cf2d3b0.png)

# 关联

- [当加入稍后读后，自动发送到你的 Telegram Channel](https://github.com/Kenshin/simpread/discussions/3092)

- [保存到 Telegraph](https://github.com/Kenshin/simpread/discussions/2795)

- [适配了一组关于 Telegram 的新站](https://github.com/Kenshin/simpread/discussions/2883)

***

> 最后感谢 @OverflowCat 的方案，另外 [保存到 Telegraph](https://github.com/Kenshin/simpread/discussions/2795) 也出自他 👏

**Comment by kchen0x** [comment]
Error handling response: TypeError: Cannot read properties of undefined (reading 'tags')

**Comment by zszwnes** [comment]
功能url已失效，望修正

**Reply by Kenshin** [reply]
@zszwnes 

okay 这个功能很久没有更新了 😂
